src/Microsoft.FeatureManagement/FeatureSettingsQueryOptions.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Options used to select a set of feature settings.
+    /// </summary>
+    class FeatureSettingsQueryOptions
+    {
+        /// <summary>
+        /// The name of an individual feature to select.
+        /// </summary>
+        public string FeatureName { get; set; }
+
+        /// <summary>
+        /// A filter used to select all feature settings after the feature with the specified name.
+        /// </summary>
+        public string After { get; set; }
```

Comment:
I am a bit hesitated with this one but don't have a good suggestion, so like to brainstorm a little :).

All we want is a way to query all feature names, but this requires that the query can be resumed from any previous query. This increases the difficulty of implementing custom `IFeatureSettingsProvider`. Even our own implementation in `ConfigurationFeatureSettingsProvider.cs` is not the easiest to understand if I am not aware of where we are from. It also makes it harder to debug if any issues are found in the future.

Reply:
The resume behavior is so that implemenations of `FeatureSettingsProvider` don't have to return all of the feature settings in one enumerable. That enumerable could be unbound.

Another way we can solve this is to have something like this

```
IFeatureSettingsProvider
{
  Task<FeatureSettings> GetFeatureSettingsAsync(string featureName);

  IAsyncEnumerable<FeatureSettings> GetAllFeatureSettingsAsync();
}
```

Async enumerable provides the chance for the implementator to page their results.

Reply:
I like the new proposal. It separates the concerns.

Reply:
Sounds good to me. I'll make the update. We'll have to depend on IAsyncEnumerable, which based off of the Azure SDK ConfigurationClient means adding a reference to [https://www.nuget.org/packages/Azure.Data.AppConfiguration/](Microsoft.Bcl.AsyncInterfaces (>= 1.0.0))

Reply:
Microsoft.Bcl.AsyncInterfaces does *not* depend on Azure SDK, right?

Reply:
Not at all. just System.Threading.Tasks.Extensions. Looks like that link in my last comment got messed up.

https://www.nuget.org/packages/Microsoft.Bcl.AsyncInterfaces/

Reply:
Looks like we should also be making this change for `GetFeatureNamesAsync`.

So it will go from

`Task<IEnumerable<string>> GetFeatureNamesAsync();`

to  

`IAsyncEnumerable<string> GetFeatureNamesAsync();`

Reply:
Makes sense. Do we like to name it `GetAllFeatureNamesAsync()`?

Reply:
I would think the plural "names" is good enough to not have the `All` part. The only reason the `All` is in IFeatureSettingsProvider is because "settings" is the same plural/singular.

Reply:
Sounds good. Agreed.